composer.json

```
@@ -24,7 +24,7 @@
     "post-update-cmd": "if php -r 'exit( version_compare( phpversion(), \"8.1\", \">=\" ) ? 0 : 1 );'; then composer --working-dir=build-cs update --no-interaction; else echo 'Skipping composer update for build-cs since not on PHP 8.1+. You are running: '; php -v; fi",
     "phpstan": "build-cs/vendor/bin/phpstan analyse --memory-limit=2048M -c phpstan.neon.dist",
     "format": "build-cs/vendor/bin/phpcbf --standard=phpcs.xml.dist --report-summary --report-source",
-    "lint": "build-cs/vendor/bin/phpcs --standard=phpcs.xml.dist",
+    "lint": "build-cs/vendor/bin/phpcs",
```

This will only result in running the `phpcs.xml.dist` file in the project root. This is ok, but we'll need to add another script to lint each plugin based on it's own ruleset.
```suggestion
    "lint": "build-cs/vendor/bin/phpcs",
    "lint:all": [
        "@lint",
        "@lint:auto-sizes"
    ],
    "lint:auto-sizes": "@lint -- ./plugins/auto-sizes --standard=./plugins/auto-sizes/phpcs.xml.dist"
```

---

Related to https://github.com/WordPress/performance/pull/995 which also removes the explicit `phpcs.xml.dist` reference in `trunk`.

---

I assume PHPCS can‘t automatically detect various config files and run them all separately?

---

Right. It only looks at parent directories to find a config file: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#using-a-default-configuration-file

It doesn't look into subdirectories for potential config overrides, as far as I know.

---

Correct, and AFAIK most tools do the same(while they can go up to the root directory to look for possible base config). They take reference to the current directory where the process is being run.

---

Thanks for all the feedback. The command has been added in https://github.com/WordPress/performance/pull/1002/commits/6b03ad06b053a637fea15d46c3570def70c8c2a9 and it works as expected. You can see the results here: https://github.com/WordPress/performance/actions/runs/8043866559/job/21966609437?pr=1002#step:9:9